Kickstarter-iOS/Views/Controllers/DiscoveryFiltersViewController.swift

```diff
@@ -60,6 +60,7 @@ internal final class DiscoveryFiltersViewController: UIViewController, UITableVi
       .observeValues { [weak self] isVisible in
         if isVisible {
           self?.dataSource.loadCategoriesLoaderRow()
+          self?.filtersTableView.reloadData()
```

Comment:
I believe there was a conscious decision to not call `reloadData` here, but I don't quite remember why, but I see that this is essentially the fix. Have you tested on a device to be sure things look ok if you first go to the list, let it load, then select a category, and come back to the filters list? I did some delaying of the reloading in this function `reloadCategories(selectedRowId: Int?)`. It's entirely possible this is ok, but I would want to be sure it's visually tested. 

Another thing you might want to try for extra safety and slightly better UX... I see in the VM that this output is called when the categories fetch terminates. Maybe a better place for the `loaderIndicatorIsVisible` output to emit `false` is when `categoriesEvent.values()` is emitted instead of on the `terminated` state of the `SignalProducer`. Since we don't handle the error state with a message or something, it seems like better feedback to just keep the loading indicator spinning if the categories never return. What do you think?